File: 단계별로 풀어보기/15. 약수, 배수와 소수 2/1735.cpp
#include <iostream>
using namespace std;

int Euclidean(int a, int b)
{
	if (b == 0)
		return a;
	else if (b == 1)
		return 1;

	return Euclidean(b, a % b);
}

int main()
{
	int a, b, c, d, denominator, numerator, LCD;
	cin >> a >> b >> c >> d;
	denominator = b * d;
	numerator = a * d + b * c;
	LCD = (denominator > numerator) ? Euclidean(denominator, numerator) : Euclidean(numerator, denominator);
	cout << numerator / LCD << " " << denominator / LCD;
	return 0;
}

File: step by step/15. Measure and Drainage/1735.py
def Euclidean(a, b):
    if b == 0:
        return a
    elif b == 1:
        return 1
    return Euclidean(b, a % b)


a, b = map(int, input().split())
c, d = map(int, input().split())
denominator = b * d
numerator = a * d + b * c

LCV = Euclidean(denominator, numerator) if denominator > numerator else Euclidean(
    numerator, denominator)
print(numerator // LCV, denominator // LCV)
